ocaml/xapi/xapi_vm.ml

```diff
@@ -505,6 +507,17 @@ let create ~__context ~name_label ~name_description
     ~nVRAM
   : API.ref_VM =
 
+  (* TODO: Error handling, but how? raise? or return null ref? or put default values? *)
+  if power_state = `Halted && suspend_VDI != Ref.null then begin
```

Comment:
you probably want structural inequality (`<>`) rather than physical inequality (`!=`)

Reply:
Indeed!